src/lib/sidebar/LeftSidebar.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+  import { BaselayerSwitcher } from "lib/common";
+  import { DefaultButton } from "lib/govuk";
+  import { map, mode, pointTool, polygonTool, routeTool } from "stores";
+  import { onDestroy } from "svelte";
+  import type { Schema } from "types";
+  import EditGeometryMode from "../draw/EditGeometryMode.svelte";
+  import { PointTool } from "../draw/point/point_tool";
+  import PointMode from "../draw/point/PointMode.svelte";
+  import { PolygonTool } from "../draw/polygon/polygon_tool";
+  import PolygonMode from "../draw/polygon/PolygonMode.svelte";
+  import RouteMode from "../draw/route/RouteMode.svelte";
+  import RouteSnapperLoader from "../draw/route/RouteSnapperLoader.svelte";
+  import SnapPolygonMode from "../draw/snap_polygon/SnapPolygonMode.svelte";
+  import StreetViewMode from "../draw/StreetViewMode.svelte";
+  import EditForm from "./EditForm.svelte";
+  import ListMode from "./ListMode.svelte";
+
+  export let schema: Schema;
+  export let routeSnapperUrl: string;
+
+  $: if ($map && !$pointTool) {
+    pointTool.set(new PointTool($map));
+  }
+  $: if ($map && !$polygonTool) {
+    polygonTool.set(new PolygonTool($map));
+  }
+
+  onDestroy(() => {
+    $pointTool?.tearDown();
+    $polygonTool?.tearDown();
+    $routeTool?.tearDown();
+  });
+</script>
+
+<div style:visibility={$mode.mode == "list" ? "visible" : "hidden"}>
+  {#if $map}
+    <RouteSnapperLoader url={routeSnapperUrl} />
+  {/if}
+</div>
+
+{#if $mode.mode == "list"}
+  <ListMode {schema} />
+{:else if $mode.mode == "edit-form"}
+  <EditForm {schema} id={$mode.id} />
+{:else if $mode.mode == "edit-geometry"}
+  <EditGeometryMode {schema} id={$mode.id} />
+{:else if $mode.mode == "new-point"}
+  <h2>New point</h2>
+  <PointMode />
+{:else if $mode.mode == "new-route"}
+  <h2>New route</h2>
+  <RouteMode />
+{:else if $mode.mode == "new-freehand-polygon"}
+  <h2>New polygon (freehand)</h2>
+  <PolygonMode />
+{:else if $mode.mode == "new-snapped-polygon"}
+  <h2>New polygon (snapped)</h2>
+  <SnapPolygonMode />
+{:else if $mode.mode == "split-route"}
+  <h2>Split route</h2>
```

Comment:
See comments in SplitRouteMode.svelte. This one is structured a little weirdly because it has to be nested underneath Map, not in this component.